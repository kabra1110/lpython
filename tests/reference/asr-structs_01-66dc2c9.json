{
    "basename": "asr-structs_01-66dc2c9",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/structs_01.py",
    "infile_hash": "3b94e692a074b226736f068daf39c876f113277a73468bd21c01d3cc",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-structs_01-66dc2c9.stdout",
    "stdout_hash": "f4abca45051c53d030a0ed049e37e9d22069b73321bc9f4adf4842ce",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}