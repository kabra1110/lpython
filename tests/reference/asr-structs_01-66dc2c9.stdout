(TranslationUnit
    (SymbolTable
        1
        {
            __main__:
                (Module
                    (SymbolTable
                        2
                        {
                            S:
                                (StructType
                                    (SymbolTable
                                        3
                                        {
                                            x:
                                                (Variable
                                                    3
                                                    x
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            y:
                                                (Variable
                                                    3
                                                    y
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    S
                                    []
                                    [x
                                    y]
                                    Source
                                    Public
                                    .false.
                                    .false.
                                    [(())
                                    (())]
                                    ()
                                    ()
                                ),
                            __main____global_statements:
                                (Function
                                    (SymbolTable
                                        6
                                        {
                                            
                                        })
                                    __main____global_statements
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    [main0]
                                    []
                                    [(SubroutineCall
                                        2 main0
                                        ()
                                        []
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            main0:
                                (Function
                                    (SymbolTable
                                        4
                                        {
                                            s:
                                                (Variable
                                                    4
                                                    s
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Struct
                                                        2 S
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    main0
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    []
                                    []
                                    [(=
                                        (Var 4 s)
                                        (StructTypeConstructor
                                            2 S
                                            [((IntegerConstant 2 (Integer 4)))
                                            (())]
                                            (Struct
                                                2 S
                                            )
                                            ()
                                        )
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                )
                        })
                    __main__
                    []
                    .false.
                    .false.
                ),
            main_program:
                (Program
                    (SymbolTable
                        7
                        {
                            __main____global_statements:
                                (ExternalSymbol
                                    7
                                    __main____global_statements
                                    2 __main____global_statements
                                    __main__
                                    []
                                    __main____global_statements
                                    Public
                                )
                        })
                    main_program
                    [__main__]
                    [(SubroutineCall
                        7 __main____global_statements
                        2 __main____global_statements
                        []
                        ()
                    )]
                )
        })
    []
)
